src/Components/Web.JS/src/Boot.WebAssembly.Common.ts

```diff
@@ -16,8 +16,9 @@ import { JSInitializer } from './JSInitializers/JSInitializers';
 import { WebAssemblyComponentDescriptor, discoverPersistedState } from './Services/ComponentDescriptorDiscovery';
 import { receiveDotNetDataStream } from './StreamingInterop';
 import { WebAssemblyComponentAttacher } from './Platform/WebAssemblyComponentAttacher';
+import { RootComponentManager } from './Services/RootComponentManager';
 
-export async function startWebAssembly(options?: Partial<WebAssemblyStartOptions>, components?: WebAssemblyComponentDescriptor[]): Promise<void> {
+export async function startWebAssembly(options?: Partial<WebAssemblyStartOptions>, components?: WebAssemblyComponentDescriptor[] | RootComponentManager): Promise<void> {
```

Comment:
Would it be practical to switch over to use `RootComponentManager` in all cases? Also on `CircuitManager.ts`.

Reply:
Possibly - the only downside I can think of is that when using `blazor.server.js` or `blazor.webassembly.js`, there would be an added interop call after app startup to add root components, rather than having them be added instantly on startup. I'm not sure if this difference in behavior is meaningful, but I thought I'd play it safe to minimize how much gets changed in this PR.